src/short_id.rs

```diff
@@ -526,3 +596,88 @@ impl From<u64> for ShortId {
 impl From<ShortId> for u64 {
     fn from(short_id: ShortId) -> Self { short_id.0 }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn short_id_is_onchain() {
+        let test_cases = vec![
+            0,
+            1,
+            100,
+            16,
+            32,
+            40,
+            0x7FFF_FF00_0000_0000,
+            0x0000_0000_0000_8000,
+        ];
+        for c in test_cases {
+            let sid = ShortId(c);
+            assert!(sid.is_onchain());
+        }
+    }
+
+    #[test]
+    fn short_id_is_offchain() {
+        let test_cases = vec![
+            0x8000_0000_0000_0000,
+            0x8000_0000_0000_0001,
+            0x9000_0000_0000_0000,
+            0xFFFF_0000_0000_0000,
+        ];
+        for c in test_cases {
+            let sid = ShortId(c);
+            assert!(sid.is_offchain());
+        }
+    }
+
+    #[test]
+    fn short_id_into() {
+        let test_cases = [0, 1];
+        for c in test_cases {
+            let sid = ShortId(c);
+            assert_eq!(sid.into_u64(), c);
+        }
+    }
+
+    #[test]
+    fn short_id_get_descriptor_empty() {
+        let sid = ShortId(0);
+        let descriptor = sid.get_descriptor();
+        match descriptor.get_block_height() {
+            Some(h) => assert_eq!(h, 0),
+            None => {}
+        }
+    }
+
+    #[test]
+    fn short_id_get_descriptor_block_height_valid() {
+        let test_cases = [
+            [0x0000_0100_0000_0000, 1],
+            [0x0000_1000_0000_0000, 16],
+            [0x0001_0000_0000_0000, 256],
+        ];
+        for c in test_cases {
+            let sid = ShortId(c[0]);
+            match sid.get_descriptor().get_block_height() {
+                Some(h) => assert_eq!(u64::from(h), c[1]),
+                None => {}
+            }
+        }
+    }
+
+    #[test]
+    #[should_panic(expected = "attempt to subtract with overflow")]
```

Comment:
This will test just a single case. To test multiple one needs to use `assert_eq` with `Err` - or `expect_err`

Reply:
Good point, it doesn't make sense to have multiple test cases here, but also cannot check the error with `assert_eq` or `expect_error` because returns an Option, no error is raised. The method panics with such input, so the only way to test it is to capture the panic with the directive.

I refactored for a single panic case in order to check panic is properly captured.

Reply:
Hm, I think the fact that you can construct incorrect ShortId and have a method call on it which will panic is an API bug. But I assume we can address it in the separate PR. Can you pls make an issue for this problem?

Reply:
Did an issue for that: https://github.com/LNP-BP/bp-core/issues/11

Reply:
lol yes code panicking definitely sounds as a bug, thanks for creating the issue, makes sense as followup :+1: